specs/hierarchical_consensus.md

```diff
@@ -0,0 +1,1082 @@
+
+The MVP implementation of this builtin-actor can be found [here](https://github.com/adlrocha/builtin-actors/tree/master/actors/hierarchical_sca). The `SCA` exposes the following functions:
+
+
+##### SCA Functions
+> Functions of the SCA. These functions are triggered when a `message` is sent for their corresponding `methodNum`.
+```go
+type SCA interface{
+    // Initializes the state of the SCA and sets
+    // all its initial parameters.
+    // (methodNum = 1)
+    Constructor(ConstructParams)
+    
+    // Register expects as source the address of the subnet actor of
+    // the subnet that wants to be registered. Its
+    // `value` an amount of collateral over the `CollateralThreshold`.
+    // This functions activates the subnet and from there on other
+    // subnets in the system are allowed to interact with it and the
+    // subnet can start commtting its checkpoints.
+    // (methodNum = 2)
+    Register()
+    
+    // AddCollateral expects as source the address of the subnet actor
+    // for which the collateral wants to be added.  Its `value` should
+    // include the amount of collateral to be added for the subnet.
+    // (methodNum = 3)
+    AddCollateral()
+    
+    // ReleaseCollateral expects as source the address of the subnet actor
+    // for which the collateral should be released. It triggers a message
+    // to the subnet actor returning the corresponding collateral. 
+    // (methodNum = 4)
+    ReleaseCollateral(value abi.TokenAmount)
+    
+    // Kill expects as source the address of the subnet actor to be killed.
+    // This function can only be executed if no collateral or circulating
+    // supply is left for the subnet (i.e. balance = 0).
+    // (methodNum = 5)
+    Kill()
+    
+    // Kill expects as source the address of the subnet actor for which the
+    // checkpoint is being committed. The function performs some basic checks
+    // to ensure that checkpoint is valid and it persist it in the SCA state.
+    // (methodNum = 6)
+    CommitChildCheckpoint(ch Checkpoint)
+    
+    // Fund can be called by any user in a subnet and it injects 
+    // the `value` of native tokens included in the message to the source's address in
+    // the child subnet given as argument.
+    // (methodNum = 7)
+    Fund(address.SubnetID)
+    
+    // Release can be called by any user in a subnet to release the amount 
+    // of native tokens included in `value` from its own address in the
+    // subnet to the address in the parent. 
+    // (methodNum = 8)
+    Release()
+    
+    // SendCross can be called by any user in the subnet to send 
+    // an arbitrary cross-message to any other subnet in
+    // the hierarchy.
+    // (methodNum = 9)
+    SendCross(msg CrossMsgParams)
+    
+    // ApplyMessage can only be called as an `ImplicitMessage` by
+    // the `SystemActor` and is used to perform the execution of a 
+    // of cross-net messages in the subnet. 
```

Comment:
Maybe it would be worth explaining what is supposed to happen in a bit more detail.

Reply:
Yes! Added an extended description of the method.